### PR TITLE
Rule Errors UI in Alerts and listing options.

### PR DESCRIPTION
**Pull request by vorillaz**
## Description

This one adds coloring and type checking for alerts listing and rule details page. This PR is the base for the upcoming ones and testing will follow at the end of the Saga.

### Screenshots:
Screenshot 2020-10-22 at 2 40 13 PM

Screenshot 2020-10-22 at 7 01 44 PM

### Checklist:

- [x] Make sure you give an high-level overview of the problem & the solution
- [x] Make sure the related issue is referenced
- [x] Make sure screenshots are added if the UI is changed in any way
- [x] ~~Make sure the changes are responsive (appearing decently on devices)~~

### CODE REVIEW

File: web/src/components/GenericItemCard/GenericItemCard.tsx [hunk 1]
```diff
@@ -40,12 +46,23 @@ interface GenericItemCardComposition {
   LineBreak: React.FC;
 }
 
-const GenericItemCard: React.FC & GenericItemCardComposition = ({ children }) => {
+const GenericItemCard: React.FC<GenericItemCardProps> & GenericItemCardComposition = ({
+  children,
+  status,
```
Comment (3nvi) [inline web/src/components/GenericItemCard/GenericItemCard.tsx:51]: given that the `GenericItemCard` is used for multiple purposes, could we rename this prop to something abstract & generic like `borderTrimColor` or `borderColor` or something like that?

In my mind I'd simply pass a color **without** associating it to something that requires context (like status)

File: web/src/pages/AlertDetails/AlertDetailsBanner/AlertDetailsBanner.tsx [hunk 2]
```diff
@@ -61,6 +51,41 @@ const AlertDetailsBanner: React.FC<AlertDetailsBannerProps> = ({ alert }) => {
           </Box>
         </Flex>
       </Flex>
+      <Flex fontSize="small-medium" pt={5}>
```
Comment (3nvi) [inline web/src/pages/AlertDetails/AlertDetailsBanner/AlertDetailsBanner.tsx:54]: optional: we can remove the `pr={8}` and replace them with a `spacing={8}` on the "parent" `<Flex>` element

**Review by 3nvi — COMMENTED** [review]
Great work @vorillaz . Couple of things to notice

- We need to update snapshot tests in the Card elements
- Let's update/write a test for the new information of the banner

Small clarification: When we have a rule error, the data oof the details tab changes. You can see this [in this mock](https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/0d293ff7-a6c6-4ddc-9dd0-442c63b91995/commits/latest/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/A57D74E1-1653-4A0C-92B7-CAA048796063?collectionId=d1ff8ec8-c4c5-462c-92e6-e2bbba4d1b89&collectionLayerId=df9127ab-bee8-4fec-a836-93be09d26e7d). A thing to notice is that the `Summaries` tab is non-existent :)

File: web/src/components/GenericItemCard/GenericItemCard.tsx (hunk 1 above)
Comment (alexmylonas) [reply web/src/components/GenericItemCard/GenericItemCard.tsx:51]: If you implement Aggelos suggestion of passing color, please add a number of specific color choices for users to pass

**Comment by vorillaz** [comment]
@3nvi You are right. Splitting the tabs for details, errors and matches are coming soon as it makes more sense to work them on a separate PR.

File: web/src/components/GenericItemCard/GenericItemCard.tsx [hunk 3]
```diff
@@ -17,9 +17,15 @@
  */
 
 import React from 'react';
-import { Box, Card, Flex, IconButton, Img, Text, TextProps } from 'pouncejs';
+import { Box, Card, Flex, IconButton, Img, Text, TextProps, Theme } from 'pouncejs';
 import { slugify } from 'Helpers/utils';
 
+export type ItemCardStatusEnum = 'pass' | 'fail';
```
Comment (3nvi) [inline web/src/components/GenericItemCard/GenericItemCard.tsx:23]: This is not used anywhere, we can delete it 😄

File: web/src/pages/AlertDetails/AlertDetailsBanner/AlertDetailsBanner.test.tsx [hunk 4]
```diff
@@ -18,13 +18,22 @@
...
 describe('AlertDetailsBanner', () => {
   it('renders', () => {
     const alert = buildAlertDetails();
+    const alertWithRuleError = buildAlertDetails({
+      type: AlertTypesEnum.RuleError,
+    });
 
-    const { container } = render(<AlertDetailsBanner alert={alert} />);
+    const { container } = render(
+      <>
```
Comment (3nvi) [inline web/src/pages/AlertDetails/AlertDetailsBanner/AlertDetailsBanner.test.tsx:32]: smart 😃

**Review by 3nvi — APPROVED** [review]
1 minor commeent, but other than that g2g 🎉

**Pull request merged**